addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/handler/XiaomiDeviceBaseHandler.java

```diff
@@ -170,24 +174,34 @@ void parseCommand(String command, JsonObject data) {
     }
 
     void parseReport(JsonObject data) {
-        logger.debug("The binding does not parse this message yet, contact authors if you want it to");
+        updateState(CHANNEL_REPORT_MSG, StringType.valueOf(data.toString()));
     }
 
     void parseHeartbeat(JsonObject data) {
-        logger.debug("The binding does not parse this message yet, contact authors if you want it to");
+        updateState(CHANNEL_HEARTBEAT_MSG, StringType.valueOf(data.toString()));
     }
 
     void parseReadAck(JsonObject data) {
-        logger.debug("The binding does not parse this message yet, contact authors if you want it to");
+        updateState(CHANNEL_READ_ACK_MSG, StringType.valueOf(data.toString()));
     }
 
     void parseWriteAck(JsonObject data) {
-        logger.debug("The binding does not parse this message yet, contact authors if you want it to");
+        updateState(CHANNEL_WRITE_ACK_MSG, StringType.valueOf(data.toString()));
     }
 
-    abstract void parseDefault(JsonObject data);
+    void parseDefault(JsonObject data) {
+        updateState(CHANNEL_LAST_MSG, StringType.valueOf(data.toString()));
+    }
 
-    abstract void execute(ChannelUID channelUID, Command command);
+    void execute(ChannelUID channelUID, Command command) {
+        if (CHANNEL_WRITE_MSG.equals(channelUID.getId())) {
+            if (command instanceof StringType) {
+                getXiaomiBridgeHandler().writeToDevice(itemId, ((StringType) command).toFullString());
+            }
+        } else {
+            logger.warn("Received command on read-only channel, thus ignoring it.");
```

Comment:
AFAIK this should not be a warn, maybe an info but In suppose a debug.

CC: @kaikreuzer

Reply:
I agree